app/src/test/java/com/example/concordia_campus_guide/HelpersTest/RoutesHelperTest/DirectionsApiDataParserTest.java

```diff
@@ -112,6 +112,49 @@ public void getDirectionsResultObj_IncorrectDataFormatTest() {
         Assert.assertNull(directionsResultObj);
     }
 
+    @Test
+    public void extractDurationAndSummaryTest() {
+        // Arrange
+        Object route = null;
+
+        DirectionsRoute directionsRoute = new DirectionsRoute();
+        directionsRoute.legs = new DirectionsLeg[1];
+        directionsRoute.legs[0] = new DirectionsLeg();
+        directionsRoute.legs[0].duration = new Duration();
+        directionsRoute.legs[0].duration.text = "10 mins";
+        directionsRoute.summary = "test";
+
+        DirectionsStep d1 = new DirectionsStep();
+        d1.duration = new Duration();
+        d1.duration.value = 142;
+        d1.travelMode = TravelMode.DRIVING;
+
+        DirectionsStep[] steps = {d1};
+
+        directionsRoute.legs[0].steps = steps;
```

Comment:
this is kind of a big method, maybe you can break down into a helper that returns the direction route u need, in the future, we might need to use it for another test and it would help to access a method that returns it to you. Or in the init().